src/dune_rules/dialect.ml

```diff
@@ -247,9 +255,12 @@ module DB = struct
           ]
     in
     let by_extension =
-      add_ext
-        (add_ext by_extension dialect.file_kinds.intf.extension)
-        dialect.file_kinds.impl.extension
+      let by_extension =
+        Option.value ~default:by_extension
```

Comment:
I think it will be clearer to just `match dialect.file_kinds.intf with` here

Reply:
done